src/ucp/api/ucp.h

```diff
@@ -2885,36 +2905,57 @@ ucs_status_t ucp_mem_advise(ucp_context_h context, ucp_mem_h memh,
  * with the memory handle the application is responsible for sharing the RKEY with
  * the peers that will initiate the access.
  *
- * @param [in]  context       Application @ref ucp_context_h "context" which was
- *                            used to allocate/map the memory.
  * @param [in]  memh          @ref ucp_mem_h "Handle" to memory region.
- * @param [out] rkey_buffer_p Memory buffer allocated by the library.
- *                            The buffer contains packed RKEY.
- * @param [out] size_p        Size (in bytes) of the packed RKEY.
+ * @param [in]  params        Memory handle packing parameters, as defined by
+ *                            @ref ucp_memh_pack_params_t.
+ * @param [out] buffer_p      Memory buffer allocated by the library.
+ *                            The buffer contains the packed memory handle.
+ * @param [out] buffer_size_p Size (in bytes) of the buffer which contains
+ *                            packed memory handle.
  *
  * @return Error code as defined by @ref ucs_status_t
  */
-ucs_status_t ucp_rkey_pack(ucp_context_h context, ucp_mem_h memh,
-                           void **rkey_buffer_p, size_t *size_p);
+ucs_status_t ucp_memh_pack(ucp_mem_h memh, ucp_memh_pack_params_t *params,
+                           void **buffer_p, size_t *buffer_size_p);
 
 
 /**
  * @ingroup UCP_MEM
- * @brief Release packed remote key buffer.
+ * @brief Memory handle pack parameters passed to @ref ucp_memh_pack.
  *
- * This routine releases the buffer that was allocated using @ref ucp_rkey_pack
- * "ucp_rkey_pack()".
+ * This structure defines the parameters that are used for releasing the
+ * UCP memory handle buffer during the @ref ucp_memh_buffer_release
+ * "ucp_memh_buffer_release" routine.
+ */
+typedef struct ucp_memh_buffer_release_params {
+    /**
+     * Mask of valid fields in this structure. All fields are mandatory.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t                field_mask;
+} ucp_memh_buffer_release_params_t;
+
+
+/**
+ * @ingroup UCP_MEM
+ * @brief Release packed memory handle buffer.
+ *
+ * This routine releases the buffer that was allocated using @ref ucp_memh_pack
+ * "ucp_memh_pack()".
  *
  * @warning
- * @li Once memory is released an access to the memory may cause a
- * failure.
+ * @li Once memory is released, an access to the memory may cause undefined
+ * behaviour.
```

Comment:
US spelling? and ditto below

Reply:
sorry, not sure that I follow this question.
shall I re-write it somehow?

Reply:
i guess it means `behaviour` -> `behavior`

Reply:
fixed